typescript/src/generator.ts

```diff
@@ -23,6 +23,9 @@ import { API } from './schema/api';
 import { processTemplates } from './templater';
 import { commonPrefix, duration } from './util';
 
+export interface Map {
```

Comment:
JavaScript already has `Map`, let's call it `OptionsMap` or something like that.

Reply:
Sounds great